homeassistant/components/rachio/binary_sensor.py

```diff
@@ -82,97 +148,64 @@ def _async_handle_any_update(self, *args, **kwargs) -> None:
         # For this device
         self._async_handle_update(args, kwargs)
 
-    @abstractmethod
-    def _async_handle_update(self, *args, **kwargs) -> None:
-        """Handle an update to the state of this sensor."""
-
-
-class RachioControllerOnlineBinarySensor(RachioControllerBinarySensor):
-    """Represent a binary sensor that reflects if the controller is online."""
-
-    _attr_device_class = BinarySensorDeviceClass.CONNECTIVITY
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique id for this entity."""
-        return f"{self._controller.controller_id}-online"
-
     @callback
     def _async_handle_update(self, *args, **kwargs) -> None:
         """Handle an update to the state of this sensor."""
-        if args[0][0][KEY_SUBTYPE] in (SUBTYPE_ONLINE, SUBTYPE_COLD_REBOOT):
-            self._attr_is_on = True
-        elif args[0][0][KEY_SUBTYPE] == SUBTYPE_OFFLINE:
-            self._attr_is_on = False
+        if self.entity_description.key == KEY_ONLINE:
+            if args[0][0][KEY_SUBTYPE] in (SUBTYPE_ONLINE, SUBTYPE_COLD_REBOOT):
+                self._attr_is_on = True
+            elif args[0][0][KEY_SUBTYPE] == SUBTYPE_OFFLINE:
+                self._attr_is_on = False
+        elif self.entity_description.key == KEY_RAIN_SENSOR:
+            if args[0][0][KEY_SUBTYPE] == SUBTYPE_RAIN_SENSOR_DETECTION_ON:
+                self._attr_is_on = True
+            elif args[0][0][KEY_SUBTYPE] == SUBTYPE_RAIN_SENSOR_DETECTION_OFF:
+                self._attr_is_on = False
```

Comment:
Let's subclass the entity description and add these as fields so we don't have to have all of the logic here

Reply:
I think that's what you were looking for, let me know if not.